src/Config.php

```diff
@@ -6,7 +6,9 @@
 
 class Config
 {
-    public function __construct(private readonly string $directory)
+    private array $cache = [];
```

Comment:
Again, caching should be removed, but this should really have been a `CacheInterface` from PSR-6 or PSR-16